st2common/st2common/bootstrap/base.py

```diff
@@ -45,3 +60,64 @@ def get_resources_from_pack(self, resources_dir):
 
         resources = sorted(resources)
         return resources
+
+    def register_packs(self, base_dirs):
+        """
+        Register packs in all the provided directories.
+        """
+        packs = self._pack_loader.get_packs(base_dirs=base_dirs)
+
+        registered_count = 0
+        for pack_name, pack_path in six.iteritems(packs):
+            self.register_pack(pack_name=pack_name, pack_dir=pack_path)
+            registered_count += 1
+
+        return registered_count
+
+    def register_pack(self, pack_name, pack_dir):
+        """
+        Register pack in the provided directory.
+        """
+        if pack_name in REGISTERED_PACKS_CACHE:
+            # This pack has already been registered during this register content run
+            return
+
+        LOG.debug('Registering pack: %s' % (pack_name))
+        REGISTERED_PACKS_CACHE[pack_name] = True
+
+        try:
+            pack_db = self._register_pack(pack_name=pack_name, pack_dir=pack_dir)
+        except Exception:
+            LOG.exception('Failed to register pack "%s"' % (pack_name))
+            return None
+
+        return pack_db
+
+    def _register_pack(self, pack_name, pack_dir):
+        """
+        Register a pack (create a DB object in the system).
+
+        Note: Pack registration now happens when registering the content and not when installing
+        a pack using packs.install. Eventually this will be moved to the pack management API.
+        """
+        manifest_path = os.path.join(pack_dir, MANIFEST_FILE_NAME)
+
+        if not os.path.isfile(manifest_path):
+            raise ValueError('Pack "%s" is missing %s file' % (pack_name, MANIFEST_FILE_NAME))
+
+        content = self._meta_loader.load(manifest_path)
+        if not content:
+            raise ValueError('Pack "%s" metadata file is empty' % (pack_name))
+
+        content['ref'] = pack_name
```

Comment:
On a related note, in the RBAC PR I will amend (and eventually replace) this with the UID.

I couldn't make that change here already because other UID changes are in a different branch and we have a chicken and the egg problem.